pkg/fleet/fleet.go

```diff
@@ -0,0 +1,41 @@
+package fleet
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetClusterHost returns the API server host for the specified client configuration.
+func GetClusterHost(clientCfg clientcmd.ClientConfig) (string, error) {
```

Comment:
Should this code instead live in Fleet and not in rancher/rancher? 
Since Fleet is always running inside the cluster `rest.InClusterConfig()`, Should always return a valid URL when running in the Fleet pod. We would have a similar flow of information where the fleet only uses `rest.InClusterConfig()` internally if it is provided with an empty string for`apiServerURL`

Reply:
No, Rancher is creating the cluster resource and needs to put a working apiServerURL inside. We don't want to have a flow where Rancher creates the cluster, events trigger, fleet decides to update the cluster with a default and more events trigger, agents need to be redeployed, etc.

When you use fleet standalone, it doe "bootstrap" the local cluster with a valid URL. However, in Rancher fleet is and its clusters are controlled by the fleetcluster controller, so they can be kept in sync with prov and management clusters.

Reply:
> We don't want to have a flow where Rancher creates the cluster, events trigger, fleet decides to update the cluster with a default and more events trigger, agents need to be redeployed, etc.

Can you elaborate on this? With these changes introduced, Rancher will deploy the fleet chart twice. Once on startup and then a second time when a user sets the `settings.ServerURL` because it will retrigger the `onSetting` handler and the URL form `InClusterConfig()` will not match the public URL set for `settings.ServerURL.` Which sounds like the scenario you want to avoid.

Reply:
We are changing our approach to this: instead of populating the API server URL in the `fleet-controller` config map, we believe it would make more sense to do so in the Kubeconfig secret used for the local cluster, which can then use that API server URL straight away instead of sourcing it from the `fleet-controller` config map.